lib/src/Base/Optim/Dlib.cxx

```diff
@@ -0,0 +1,862 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Dlib solvers
+ *
+ *  Copyright 2005-2019 Airbus-EDF-IMACS-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "openturns/Dlib.hxx"
+#include "openturns/DlibFunctions.hxx"
+#include "openturns/SpecFunc.hxx"
+#include "openturns/PersistentObject.hxx"
+
+#ifdef OPENTURNS_HAVE_DLIB
+  #include <dlib/optimization.h>
+  #ifdef OPENTURNS_HAVE_DLIB_GLOBAL_OPTIMIZATION
+    #include <dlib/global_optimization.h>
+  #endif
+#endif
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/** DEFINITION OF SEARCH STRATEGY CLASSES **/
+class DlibSearchStrategyImplementation
+  : public PersistentObject
+{ 
+  public:
+    DlibSearchStrategyImplementation()
+    : PersistentObject()
+    , wolfeRho_(ResourceMap::GetAsScalar("Dlib-DefaultWolfeRho"))
+    , wolfeSigma_(ResourceMap::GetAsScalar("Dlib-DefaultWolfeSigma"))
+    , maxLineSearchIterations_(ResourceMap::GetAsUnsignedInteger("Dlib-DefaultMaxLineSearchIterations"))
+    {
+      // Nothing to do
+    }
+    
+    DlibSearchStrategyImplementation(const Scalar wolfeRho,
+                                     const Scalar wolfeSigma,
+                                     const UnsignedInteger maxLineSearchIterations )
+    : PersistentObject()
+    , wolfeRho_(wolfeRho)
+    , wolfeSigma_(wolfeSigma)
+    , maxLineSearchIterations_(maxLineSearchIterations)
+    {
+      // Check wolfeRho and wolfeSigma
+      if (!(wolfeRho < wolfeSigma)) throw InvalidArgumentException(HERE) << "Error: wolfeRho must be lower than wolfeSigma";
+    }
+    
+      // Virtual constructor
+    virtual DlibSearchStrategyImplementation * clone() const
+    {
+      return new DlibSearchStrategyImplementation(*this);
+    }
+      
+    /** ACCESSORS **/
+    virtual double get_wolfe_rho()
+    {
+      return wolfeRho_;
+    }
+    virtual double get_wolfe_sigma()
+    {
+      return wolfeSigma_;
+    }
+    virtual unsigned long get_max_line_search_iterations ()
+    {
+      return maxLineSearchIterations_;
+    }
+        
+    /** FUNCTION USED IN ACTUAL SEARCH **/
+    virtual const DlibMatrix get_next_direction ( const DlibMatrix& x,
```

Comment:
getNextDirection, and add a space before &

Reply:
same